google-cloud-logging/lib/google/cloud/logging/async_writer.rb

```diff
@@ -13,21 +13,23 @@
 # limitations under the License.
 
 
-require "set"
-require "stackdriver/core/async_actor"
+require "monitor"
+require "concurrent"
+require "google/cloud/logging/errors"
 
 module Google
   module Cloud
     module Logging
       ##
       # # AsyncWriter
       #
-      # An object that batches and transmits log entries asynchronously.
+      # AsyncWriter buffers, batches, and transmits log entries efficiently.
+      # Writing log entries is asynchronous and will not block.
       #
-      # Use this object to transmit log entries efficiently. It keeps a queue
-      # of log entries, and runs a background thread that transmits them to
-      # the logging service in batches. Generally, adding to the queue will
-      # not block.
+      # Batches that cannot be delivered immediately are queued. When the queue
+      # is full new batch requests will raise errors that can be consumed using
+      # the {#on_error} callback. This provides back pressure in case the writer
+      # keep up with requests.
```

Comment:
Something wrong with the grammar here: "in case the writer keep up with requests." While you are fixing this I would love to hear a bit more about how this backpressure mechanism works. Errors "can be consumed" -- what does this mean in practice? The `#on_error` example shows `puts error`. I'm wondering what I really should be doing. (I still haven't made it through all of the docs, however.)

Reply:
I assume "in case the writer cannot keep up with requests" is meant.

Reply:
Correct. This should be "the writer cannot keep up with the requests".

Reply:
The other code examples that demonstrate `on_error` show using Error Reporting to report the error. This code example was missed. I'll update it.

```ruby
# Register to be notified when unhandled errors occur.
async.on_error do |error|
  # error can be a AsyncWriterError, with entries
  Google::Cloud::ErrorReporting.report error
end
```

Reply:
The text and code example are update. PTAL and resolve this conversation if satisfied.